src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -46,6 +46,68 @@ def boolified_type(symbol_type: ct.SymbolType) -> ct.ScalarType | ct.FieldType:
     raise GTTypeError(f"Can not boolify type {symbol_type}!")
 
 
+def construct_tuple_type(
+    element_types: list[ct.TupleType],
```

Comment:
Is there any case where `element_types` does not refer to the types in the "true-branch"? If not, would you agree that there would be some benefit in renaming?

Reply:
agree and edited